src/main/kotlin/com/dansplugins/factionsystem/dynmap/MfDynmapService.kt

```diff
@@ -51,265 +76,170 @@ class MfDynmapService(private val plugin: MedievalFactions) {
-        val claimService = plugin.services.claimService
-        val playerService = plugin.services.playerService
-        return buildString {
-            append("<h1>${faction.name}</h1>")
-            append("<h2>Description</h2>")
-            append(faction.description)
-            append("<h2>Members (${faction.members.size})</h2>")
-            append(
-                faction.members.groupBy { it.role }.map { (role, members) ->
-                    """
-                        <h3>${role.name} (${faction.members.count { it.role.id == role.id }})</h3>
-                        ${members.joinToString { member -> playerService.getPlayer(member.playerId)?.name ?: plugin.language["UnknownPlayer"] }}
-                    """.trimIndent()
-                }.joinToString("<br />")
-            )
-            val liegeId = relationshipService.getLiege(faction.id)
-            val liege = liegeId?.let(factionService::getFaction)
-            if (liege != null) {
-                append("<h2>Liege</h2>")
-                append(liege.name)
-                append("<br />")
-            }
-            val vassals = relationshipService.getVassals(faction.id).mapNotNull(factionService::getFaction)
-            if (vassals.isNotEmpty()) {
-                append("<h2>Vassals</h2>")
-                append(vassals.joinToString(transform = MfFaction::name))
-                append("<br />")
-            }
-            val allies = relationshipService.getAllies(faction.id).mapNotNull(factionService::getFaction)
-            if (allies.isNotEmpty()) {
-                append("<h2>Allies</h2>")
-                append(allies.joinToString(transform = MfFaction::name))
-                append("<br />")
-            }
-            val atWarWith = relationshipService.getFactionsAtWarWith(faction.id).mapNotNull(factionService::getFaction)
-            if (atWarWith.isNotEmpty()) {
-                append("<h2>At war with</h2>")
-                append(atWarWith.joinToString(transform = MfFaction::name))
-                append("<br />")
+        val realm = claimService.getClaims(faction.id) + relationshipService.getVassalTree(faction.id).flatMap(claimService::getClaims)
+        realm.groupBy { it.worldId }.forEach { (worldId, worldClaims) ->
+            createUpdateTask(faction.id, { updateWorldRealm(faction, worldId, worldClaims, realmMarkerSet) }) { runTask(plugin) }
+        }
+    }
+
+    private fun updateWorldRealm(faction: MfFaction, worldId: UUID, worldClaims: List<MfClaimedChunk>, realmMarkerSet: MarkerSet) {
+        val world = plugin.server.getWorld(worldId)
+        if (world != null) {
+            createUpdateTask(faction.id, { createRealmMarkers(faction, world, worldClaims, realmMarkerSet) }) { runTaskAsynchronously(plugin) }
+        }
+    }
+
+    private fun createRealmMarkers(faction: MfFaction, world: World, worldClaims: List<MfClaimedChunk>, realmMarkerSet: MarkerSet) {
+        val paths = ClaimPathBuilder.getPaths(worldClaims)
```

Comment:
Rather than instantiate a new instance of ClaimPathBuilder every time this method is called we should use a class instance

Reply:
Addressed